packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -124,6 +126,10 @@ class ButtonBase extends React.Component {
     clearTimeout(this.focusVisibleTimeout);
   }
 
+  getButtonNode() {
+    return ReactDOM.findDOMNode(this.buttonRef.current);
```

Comment:
What's the advantage of calling `ReactDOM.findDOMNode` every time we need it? Why not only calling it and storing it during the `componentDidMount` lifecycle?

Reply:
> Because that would assume that during the lifetime of the component the child component to which we attach the ref does not change its DOM node.

@eps1lon I have been hit multiple time by this issue in the past, most recently with the tooltip and component creation in the render method. The soluion was to fix the RootRef component to correctly forward ref dom change updates. Ok, let's move with that.

Reply:
It's also leaky as far as I understand my own codesandbox. The old DOM node can't be garbage collected since we still hold a reference to it in the parent component. It's very niche for sure but I think that we should handle that. At least for components with proper forwarded refs this has no perf implications beyond a function call and 1 or 2 strict equality checks. For everyone else it depends on the depth (and it looks like breadth too) of the child component tree.